### ISSUE TITLE
dxcdisassembler appends raw integer to Texture2DMSArray Dimension during disassembly

### ISSUE THREAD

**Issue by spacehamster** [microsoft/DirectXShaderCompiler#3335]
The line https://github.com/microsoft/DirectXShaderCompiler/blob/d574d27c1fd49333cc7a1280dfccd840523d87aa/tools/clang/tools/dxcompiler/dxcdisassembler.cpp#L422

appends a unsigned to a string without processing when it should first convert the unsigned to a string before the addition.

As a result, when Texture2DMSArrays such as `Texture2DMS<float3,8>` are disassembled, the output is 
```
; Resource Bindings:
;
; Name                                 Type  Format         Dim      ID      HLSL Bind  Count
; ------------------------------ ---------- ------- ----------- ------- -------------- ------
; srv1                              texture     f32       2dMS\b      T0             t3     1
```
instead of
```
; Resource Bindings:
;
; Name                                 Type  Format         Dim      ID      HLSL Bind  Count
; ------------------------------ ---------- ------- ----------- ------- -------------- ------
; srv1                              texture     f32       2dMS8      T0             t3     1
```
This can be seen in the testcase https://github.com/microsoft/DirectXShaderCompiler/blob/master/tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/srv_ms_load2.hlsl

**Comment by pow2clk** [comment]
I have a fix pending #3473. 

While it's a joy to fix bugs that have already been root caused, we welcome fixes like this from anyone if you'd care to submit a PR of your own 😄

**Comment by spacehamster** [comment]
Thank you for the fix. I had intended to submit a PR, but I got distracted between opening the issue and actually setting up the environment and writing the fix and completely forgot about it 😳
